recipes/zulu-openjdk/all/conanfile.py

```diff
@@ -16,30 +16,33 @@ class ZuluOpenJDK(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
 
     @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
+    def _settings_host(self):
+        return getattr(self, "settings_host", self.settings)
 
     @property
     def _jni_folder(self):
-        folder = {"Linux": "linux", "Macos": "darwin", "Windows": "win32"}.get(str(self._settings_build.os))
+        folder = {"Linux": "linux", "Macos": "darwin",
+                  "Windows": "win32", "SunOS": "solaris"}.get(str(self._settings_host.os))
         return os.path.join("include", folder)
 
     def package_id(self):
         del self.info.settings.compiler
         del self.info.settings.build_type
 
     def validate(self):
-        supported_archs = ["x86_64", "armv8"]
-        if self._settings_build.arch not in supported_archs:
-            raise ConanInvalidConfiguration(f"Unsupported Architecture ({self._settings_build.arch}). "
-                                            f"This version {self.version} currently only supports {supported_archs}.")
-        supported_os = ["Windows", "Macos", "Linux"]
-        if self._settings_build.os not in supported_os:
-            raise ConanInvalidConfiguration(f"Unsupported os ({self._settings_build.os}). "
-                                            f"This package currently only support {supported_os}.")
+        srcs = self.conan_data["sources"].get(self.version, {}).get(str(self._settings_host.os))
+        if srcs is None:
+            raise ConanInvalidConfiguration(f"Unsupported os ({self._settings_host.os}). "
+                                            f"This version {self.version} currently does not support"
+                                            f" {self._settings_host.arch} on {self._settings_host.os})")
```

Comment:
```suggestion
                                            f" {self.settings.arch} on {self.settings.os})")
```
You can use `self.settings` instead (here and elsewhere), which corresponds to host-context settings and is always available.